azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/ClientInfo.java

```diff
@@ -0,0 +1,35 @@
+package com.microsoft.azure.eventhubs;
+
+import com.microsoft.azure.eventhubs.amqp.AmqpConstants;
+import org.apache.qpid.proton.amqp.Symbol;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ClientInfo {
+    private String productName = ClientConstants.PRODUCT_NAME;
+
+    private String productVersion = ClientConstants.CURRENT_JAVACLIENT_VERSION;
+
+    public static ClientInfo Default = new ClientInfo(null, null);
+
+    public ClientInfo(final String productName, final String productVersion) {
+        if (productName != null) {
```

Comment:
nit: consider this pattern if you like :)

private final String productName;
public ClientInfo(final String productName, final String productVersion) { 
this.productName = productName == null ? constants.Product_name : ...;

and then getConnProperties just initialize once lazily - whenever it gets called first.

Reply:
I didn't get the 'getConnProperties' suggestion. Should I cache the map as a class member and initialize it the first time the method is called?